File: SIN BDD/SOAP_DOTNET_CONUNI_GR06/08. SERVIDOR/servidor_soap_dotnet_conuni_gr06/Pruebas/PruebaTemperatura.cs
using Ec.Edu.Monster.Controlador;

namespace Ec.Edu.Monster.Pruebas;

static class PruebaTemperatura
{
    static readonly CONUNI servicio = new();
    static int ok = 0, fail = 0;

    public static void Ejecutar()
    {
        Console.WriteLine("=== PRUEBAS DE CONVERSION DE TEMPERATURA ===");
        Console.WriteLine();

        Probar("celsiusAFahrenheit", 0, 32);
        Probar("fahrenheitACelsius", 32, 0);
        Probar("celsiusAKelvin", 0, 273.15);
        Probar("kelvinACelsius", 273.15, 0);
        Probar("fahrenheitAKelvin", 32, 273.15);
        Probar("kelvinAFahrenheit", 273.15, 32);

        Console.WriteLine();
        Console.WriteLine($"--- RESULTADOS TEMPERATURA: OK={ok} FAIL={fail} ---");
        Console.WriteLine();
    }

    static void Probar(string operacion, double valor, double esperado)
    {
        try
        {
            double obtenido = operacion switch
            {
                "celsiusAFahrenheit" => servicio.CelsiusAFahrenheit(valor),
                "fahrenheitACelsius" => servicio.FahrenheitACelsius(valor),
                "celsiusAKelvin" => servicio.CelsiusAKelvin(valor),
                "kelvinACelsius" => servicio.KelvinACelsius(valor),
                "fahrenheitAKelvin" => servicio.FahrenheitAKelvin(valor),
                "kelvinAFahrenheit" => servicio.KelvinAFahrenheit(valor),
                _ => throw new ArgumentOutOfRangeException(nameof(operacion))
            };

            bool casoOk = Math.Abs(obtenido - esperado) <= ConstantesPruebas.Tolerancia;

            if (casoOk)
            {
                ok++;
                Console.WriteLine($"[OK] {operacion}({valor}) = {obtenido:F4}");
            }
            else
            {
                fail++;
                Console.WriteLine($"[FAIL] {operacion}({valor})");
                Console.WriteLine($"  Esperado: {esperado} +/- {ConstantesPruebas.Tolerancia}");
                Console.WriteLine($"  Obtenido: {obtenido}");
            }
        }
        catch (Exception e)
        {
            fail++;
            Console.WriteLine($"[FAIL] {operacion}({valor}) -> Error: {e.Message}");
        }
    }
}

File: SIN BDD/RESTFULL_JAVA_CONUNI_GR06/08. SERVIDOR/servidor_restful_java_conuni_gr06/test/ec/edu/monster/prueba/pruebaTemperatura.java
package ec.edu.monster.prueba;

import ec.edu.monster.servicio.ServicioTemperatura;
import org.junit.Test;
import static org.junit.Assert.*;

public class pruebaTemperatura {

    private final ServicioTemperatura servicio = new ServicioTemperatura();
    private static final double MARGEN = 0.0001;

    @Test
    public void pruebaCelsiusAFahrenheit() {
        assertEquals(32.0, servicio.celsiusAFahrenheit(0.0), MARGEN);
        assertEquals(212.0, servicio.celsiusAFahrenheit(100.0), MARGEN);
    }

    @Test
    public void pruebaFahrenheitACelsius() {
        assertEquals(0.0, servicio.fahrenheitACelsius(32.0), MARGEN);
        assertEquals(100.0, servicio.fahrenheitACelsius(212.0), MARGEN);
    }

    @Test
    public void pruebaCelsiusAKelvin() {
        assertEquals(273.15, servicio.celsiusAKelvin(0.0), MARGEN);
        assertEquals(373.15, servicio.celsiusAKelvin(100.0), MARGEN);
    }

    @Test
    public void pruebaKelvinACelsius() {
        assertEquals(0.0, servicio.kelvinACelsius(273.15), MARGEN);
        assertEquals(-273.15, servicio.kelvinACelsius(0.0), MARGEN);
    }

    @Test
    public void pruebaFahrenheitAKelvin() {
        assertEquals(273.15, servicio.fahrenheitAKelvin(32.0), MARGEN);
        assertEquals(373.15, servicio.fahrenheitAKelvin(212.0), MARGEN);
    }
}
